src/t8_forest/t8_forest_cxx.cxx

```diff
@@ -1929,6 +1929,7 @@ t8_forest_leaf_face_neighbors (t8_forest_t forest, t8_locidx_t ltreeid,
         element_index =
           t8_forest_bin_search_lower (element_array, neigh_id,
                                       forest->maxlevel);
+        T8_ASSERT (element_index >= 0);
```

Comment:
The fixed volume bug seems to have had an influence on this issue, but it is still not running. Now this assertion is triggering

Reply:
Thanks.
In which time step did this occur and where you computing in parallel?

I am currently having a serial run that is in time step 6 and no assertion has occured so far.

Reply:
I have merged with main which did a lot of updates, maybe this fixed the error that you got?

Reply:
Update: Now in time step 20 and still going.
(due to debugging mode being on it is rather slow)

Reply:
I do not know the time step, but the assertion triggered after less than 10 sec of runtime in debug mode (I called `mpirun -np 8 --oversubscribe ./t8_advection -u7 -l0 -r1 -f cube -d3 -C0.5 -b4 -X-0.7 -Y0 -Z0 -R0.3 -T1`, `--oversubscribe` was needed due to me using OpenMPI)

Reply:
But with `mpirun -np 4 ./t8_advection -u7 -l0 -r1 -f cube -d3 -C0.5 -b4 -X-0.7 -Y0 -Z0 -R0.3 -T1` it runs smoothly

Reply:
Traceback:
```
[libsc 6] Abort: Assertion 'element_index >= 0'
[libsc 6] Abort: ./../t8code/src/t8_forest/t8_forest_cxx.cxx:1932
[libsc 6] Abort: Obtained 10 stack frames
[libsc 6] Stack 0: t8_advection(+0xc97a5) [0x558f0443b7a5]
[libsc 6] Stack 1: t8_advection(+0xc9760) [0x558f0443b760]
[libsc 6] Stack 2: t8_advection(+0xc9981) [0x558f0443b981]
[libsc 6] Stack 3: t8_advection(+0x5c539) [0x558f043ce539]
[libsc 6] Stack 4: t8_advection(+0x18a3c) [0x558f0438aa3c]
[libsc 6] Stack 5: t8_advection(+0x1963e) [0x558f0438b63e]
[libsc 6] Stack 6: t8_advection(+0x1b256) [0x558f0438d256]
[libsc 6] Stack 7: libc.so.6(+0x29d90) [0x7f6bd6891d90]
[libsc 6] Stack 8: libc.so.6(__libc_start_main+0x80) [0x7f6bd6891e40]
[libsc 6] Stack 9: t8_advection(+0x15d25) [0x558f04387d25]
```
Lines:
```
addr2line -e ./t8_advection -i
0x1b256
/home/elsw_sa/git/t8code_debug/./../t8code/example/advect/t8_advection.cxx:1851
0x1963e
/home/elsw_sa/git/t8code_debug/./../t8code/example/advect/t8_advection.cxx:1371
0x18a3c
/home/elsw_sa/git/t8code_debug/./../t8code/example/advect/t8_advection.cxx:1166
0x5c539
/home/elsw_sa/git/t8code_debug/./../t8code/src/t8_forest/t8_forest_cxx.cxx:1935
0xc9981
/home/elsw_sa/git/t8code_debug/sc/.././../t8code/sc/src/sc.c:1044
0xc9760
/home/elsw_sa/git/t8code_debug/sc/.././../t8code/sc/src/sc.c:982
0xc97a5
/home/elsw_sa/git/t8code_debug/sc/.././../t8code/sc/src/sc.c:997
```

Reply:
This is not solved. But close the conversation for this PR.
See #390, which remains open.